### PR TITLE
Replace maestro test token for GitHub

### PR DESCRIPTION
**Pull request by oleksandr-didyk**
Resolves https://github.com/dotnet/arcade-services/issues/4081

Maestro build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2566239&view=results
PCS build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2566241&view=results

### CODE REVIEW

File: eng/templates/jobs/e2e-pcs-tests.yml [hunk 1]
```diff
@@ -106,7 +104,7 @@ jobs:
     env:
       PCS_BASEURI: ${{ variables.PcsTestEndpoint }}
       PCS_TOKEN: $(GetAuthInfo.Token)
-      GITHUB_TOKEN: $(maestro-scenario-test-github-token)
```
Comment (premun) [inline eng/templates/jobs/e2e-pcs-tests.yml:109]: Should it get removed from some secret manifests? And some KeyVaults?

File: azure-pipelines-product-construction-service.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ pr:
 
 variables:
 # https://dev.azure.com/dnceng/internal/_library?itemType=VariableGroups&view=VariableGroupView&variableGroupId=189
+# Required for MaestroAppClientId, MaestroStagingAppClientId
```
Comment (premun) [inline azure-pipelines-product-construction-service.yml:15]:
```suggestion
# Required for MaestroAppClientId, MaestroStagingAppClientId, BotAccount-dotnet-bot-repo-PAT
```

**Review by premun — CHANGES_REQUESTED** [review]
@oleksandr-didyk we should not use the PAT you're using. That PAT has access to all GitHub repos under `dotnet/`. But the E2E tests are targeting `maestro-auth-test` so this should be a separate PAT for that org. I am not even sure this PAT would work.

File: eng/templates/jobs/e2e-pcs-tests.yml (hunk 1 above)
Comment (oleksandr-didyk) [reply eng/templates/jobs/e2e-pcs-tests.yml:109]: If you remember its not managed in any of our secret manifest.

As for Key Vaults - I'll remote it once I remove it from release CLI as its referenced there

**Comment by oleksandr-didyk** [comment]
> @oleksandr-didyk we should not use the PAT you're using. That PAT has access to all GitHub repos under `dotnet/`. But the E2E tests are targeting `maestro-auth-test` so this should be a separate PAT for that org. I am not even sure this PAT would work.

Oh, that's a good point. I did work as both Maestro and PCS builds passed and I see the PRs cleaned up. For example: https://github.com/maestro-auth-test/maestro-test2/pull/23959

I will generate a new PAT for the test org specifically then

**Comment by oleksandr-didyk** [comment]
Discussed over Teams, we would benefit from generating a new token not only for the tests, but also for the other actions the pipeline performs on GitHub, such as creating releases in `arcade-services`.

Will close this PR for now and re-open with the new token in place

**Pull request closed without merging**